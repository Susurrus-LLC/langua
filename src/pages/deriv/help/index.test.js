import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureDerivHelp as DerivHelp } from '.'

describe('Deriv', () => {
  const data = {
    site: {
      siteMetadata: {
        title: 'Langua',
        toolInfo: {
          deriv: {
            title: 'Deriv',
            link: 'deriv',
            description: 'Deriv description'
          }
        }
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<DerivHelp data={data} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    let tree = renderer.create(<DerivHelp data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
