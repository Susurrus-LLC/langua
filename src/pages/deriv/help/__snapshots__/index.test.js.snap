// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deriv renders accurately 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <div
      className="titleArea"
    >
      <h1
        className="siteTitle"
      >
        <a
          className="titleLink"
          href="/"
        />
      </h1>
      <p
        className="subtitle"
      />
    </div>
    <nav
      className="nav"
    >
      <ul
        className="menu"
      >
        <li
          className="navLink"
        >
          <a
            href="/gen"
          >
            Gen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/morph"
          >
            Morph
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/deriv"
          >
            Deriv
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/frequen"
          >
            Frequen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/tree"
          >
            Tree
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/lex"
          >
            Lex
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <a
      className="btn buttonLink"
      href="/deriv"
    >
      Back
    </a>
    <h2
      className="toolTitle"
    >
      undefinedDeriv Help
    </h2>
    <p
      id="description"
    >
      undefinedDeriv Deriv description
    </p>
    <h3
      id="using"
    >
      Using 
      Langua
      Deriv
    </h3>
    <h4
      id="lexemes"
    >
      Lexemes
    </h4>
    <p>
      Enter a list of all lexemes that should be used as roots to be derived from. Also add a brief gloss for each of the lexemes.
    </p>
    <h4
      id="derivations"
    >
      Derivational Affixes
    </h4>
    <p>
      Enter a list of all derivational affixes that should be used to derive new words. Each gloss should contain a single hyphen that marks the point at which the affix attaches to the root. If the hyphen is placed at the start, the affix will be a suffix. If the hyphen is placed at the end, the affix will be a prefix. If the hyphen is placed within the affix, it will be split and attached as a circumfix on either side of the root. Add a gloss for each of the affixes and a brief definition of the change the affix creates.
    </p>
    <h4
      id="derive"
    >
      Derive
    </h4>
    <p>
      When the list of lexemes and affixes is ready, click the Derive button to generate a list of roots with possible derivational affixes added to them. Hovering over the affix glosses in the results will display the affix’s definition below that gloss.
    </p>
    <p>
      There are also a couple adjustments you can make. First, you can choose the total number of words that should be derived from the input. The tool will always filter out duplicates, so there will never be more results than are mathematically possible given the input. Second, you can choose the type of output that should be generated. You can choose to use just a single root and generate the chosen number of words using different affixes on that root, you can choose to use just a single affix and generate the chosen number of words by attaching it to different roots, or you can choose to randomly combine any root with any affix.
    </p>
    <p>
      After the derivations have been generated, some statistics are shown below, including how many derivations were printed to the output and how many derivations are possible based on the number of lexemes and affixes provided.
    </p>
    <h4
      id="files"
    >
      Saving and Loading Settings
    </h4>
    <p>
      Clicking the 
      <strong>
        Save
      </strong>
       button will save the current settings to the browser’s local storage and generate a small
       
      <code>
        .lngd
      </code>
       text file containing the current settings that can be saved to your disk. This 
      <code>
        .lngd
      </code>
       file can be loaded using the
       
      <strong>
        Open
      </strong>
       button to reload saved settings.
    </p>
    <h3
      id="acknowledgments"
    >
      Acknowledgments
    </h3>
    <p>
      Much thanks should be given to Jan Strasser and
       
      <a
        href="http://akana.conlang.org/tools/derivizer.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        the Derivizer
      </a>
      . 
      Langua
      Deriv
       was mainly built as a modernized and updated version of the Derivizer.
    </p>
    <div
      className="Toastify"
    />
  </main>
  <footer
    className="footer"
  >
    <p
      className="copyright"
    >
      <a
        href="undefined/blob/master/CHANGELOG.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        Version 
        <i
          className="fas fa-code"
        />
         
      </a>
      . Built by
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      />
      , copyright © 
      2017–2020
       under the
       
      <a
        href="https://choosealicense.com/licenses/agpl-3.0/"
        rel="noopener noreferrer"
        target="_blank"
      >
        AGPL-3.0 
        <i
          className="fas fa-gavel"
        />
         license
      </a>
      .
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        View this project on 
        <i
          className="fab fa-github"
        />
         GitHub.
      </a>
    </p>
  </footer>
</div>
`;
